_sources/GeneralIntro/Algorithms.rst

```diff
@@ -11,15 +11,70 @@
    :prefix: intro-2-
    :start: 1
 
+.. index:: programming language, algorithm, pseudocode,
+           program
+
 Algorithms
 ----------
+           
+If problem solving is a central part of computer science, then the solutions that you create through
+the problem solving process are also important.  In computer science, we refer to these solutions
+as **algorithms**.  An algorithm is a step by step list of instructions that if followed exactly will solve the problem under consideration.
+
+For example, an algorithm to sort a list of numbers in order from least to greatest might look like this:
+
+.. admonition:: Algorithm Example 1 (English)
+
+    1. Ask for radius 
+    2. Compute area by squaring radius and multiplying the result by pi
+    3. Display the computed area 
+
+Notice that this algorithm consists of a set of numbered steps. It is written in English, for ease of 
+understanding. Although simple algorithms are easily understood when written in English, more complicated
+algorithms need more precise notation. For improved precision, algorithms are often written in pseudocode. **Pseudocode** is
+a notation that is more precise than English but generally not as precise as a programming language.
+The same algorithm expressed in pseudocode might look something like this:
+
+.. admonition:: Algorithm Example 2 (Pseudocode)
+
+    1. Ask for radius 
+    2. let area = (radius\ :sup:`2`) × π
+    3. Display area 
+
+Note how the pseudocode example expresses step 2 more precisely, specifying the formula in mathematical
+terms.
+
+Our goal in computer science is to take a problem and develop an algorithm that can serve as a general solution.  
+Once we have such a solution, we can use our computer to automate its execution using programming. 
+Programming is a skill that allows a computer scientist to take an algorithm and represent it in
+a notation (a program) that can be followed by a computer.  A program is written in a **programming language**
+such as Python, the language you will learn in this book.
+
+To help you understand the difference between an algorithm and a program, consider this program to compute
+the area of a circle:
+
+.. activecode:: alg_impl
+   :nocodelens:
+
+   radius = int(input("Enter the radius:"))
+   area = (radius * radius) * 3.1415
+   print("The area of a circle with radius", radius, "is:", area)
+
+A **program** is an algorithm expressed in a programming language. We might also say
+that a program is an *implementation* of an algorithm. In this example, both the
+algorithm and the program have three steps. The first step gets some input from
+the user; the second step performs a calculation using the information obtained
```

Comment:
and transforms the input into something that the computer can do math with??

Reply:
I incorporated your suggestion.